codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlMemberDeserVisitor.java

```diff
@@ -106,9 +106,22 @@ private void consumeToken() {
 
         writer.write("val, done, err := decoder.Value()");
         writer.write("if err != nil { return err }");
+        writer.openBlock("if done {", "}", () -> {
+            handleDone(shape);
+            writer.write("break");
+        });
+    }
 
-        // Since we got the matching end element we skip setting the value
-        writer.write("if done { break }");
+    // handles zero value assignment in case a empty xml tag is sent for the response member
+    private void handleDone(Shape shape) {
+        GoWriter writer = context.getWriter();
+
+        // Only string shapes can have an empty xml tag in response. If an empty xml tag is
+        // found for a string shape, we deserialize and the member value to a empty string.
+        if (shape.isStringShape() && !shape.hasTrait(EnumTrait.ID)) {
```

Comment:
Blob shapes can also be empty. Are collection shapes transparently handled? I'd imagine so.

Reply:
Blobs will be handled correctly by this change. Collection shapes are handled already. 

I have tested them with protocol tests added in https://github.com/awslabs/smithy/pull/547

Reply:
The first line of your comment is still technically wrong. It might be good to call out that that those other cases are handled and mention where.